homeassistant/components/withings/__init__.py

```diff
@@ -59,52 +85,90 @@ async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
-                "profile": data.get("profile"),
-                "token": {
-                    "access_token": creds.get("access_token"),
-                    "refresh_token": creds.get("refresh_token"),
-                    "expires_at": int(creds.get("token_expiry")),
-                    "type": creds.get("token_type"),
-                    "userid": creds.get("userid") or creds.get("user_id"),
-                },
-            },
-        )
-
+    # Create the date manager.
     implementation = await config_entry_oauth2_flow.async_get_config_entry_implementation(
         hass, entry
     )
+    data_manager = init_data_manager(
+        hass, entry, implementation, hass.data[DOMAIN][const.CONFIG]
+    )
 
-    data_manager = get_data_manager(hass, entry, implementation)
+    # Add the data manager to the push handler so the data manager can
+    # be updated when withings pushes data to HA.
+    webhook_processor = get_webhook_processor(hass)
+    webhook_processor.add_data_manager(data_manager)
 
-    _LOGGER.debug("Confirming we're authenticated")
+    _LOGGER.debug("Confirming %s is authenticated to withings", data_manager.profile)
     try:
-        await data_manager.check_authenticated()
+        await data_manager.async_check_authenticated()
     except NotAuthenticatedError:
-        _LOGGER.error(
-            "Withings auth tokens exired for profile %s, remove and re-add the integration",
-            data_manager.profile,
-        )
+        data_manager.show_auth_notification()
         return False
 
-    hass.async_create_task(
-        hass.config_entries.async_forward_entry_setup(entry, "sensor")
-    )
+    async def start_platforms():
+        await hass.config_entries.async_forward_entry_setup(entry, BINARY_SENSOR_DOMAIN)
+        await hass.config_entries.async_forward_entry_setup(entry, SENSOR_DOMAIN)
+        data_manager.start_polling()
+
+    hass.async_create_task(start_platforms())
 
     return True
 
 
-async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Withings config entry."""
-    return await hass.config_entries.async_forward_entry_unload(entry, "sensor")
+    await async_remove_data_manager(hass, entry)
+
+    await hass.config_entries.async_forward_entry_unload(entry, BINARY_SENSOR_DOMAIN)
+    await hass.config_entries.async_forward_entry_unload(entry, SENSOR_DOMAIN)
+
+    return True
+
+
+async def async_webhook_handler(
+    hass: HomeAssistant, webhook_id: str, request: Request
+) -> Response:
+    """Handle webhooks calls."""
+    # Handle http head calls to the path.
+    # When creating a notify subscription, Withings will check that the endpoint is running by sending a HEAD request.
+    if request.method.upper() == "HEAD":
+        return Response(headers={"ETag": async_create_etag()})
+
+    # Handle http post calls to the path.
+    if request.method.upper() == "POST":
```

Comment:
Use a guard clause.

```python
if request.method.upper() != "POST":
    return
```

Reply:
When you add the guard clause, you can remove this line because it's already checked :)